api4/channel.go

```diff
@@ -88,6 +92,97 @@ func createDirectChannel(c *Context, w http.ResponseWriter, r *http.Request) {
 	}
 }
 
+func getChannel(c *Context, w http.ResponseWriter, r *http.Request) {
+	c.RequireChannelId()
+	if c.Err != nil {
+		return
+	}
+
+	allowed := false
+
+	if !allowed && !app.SessionHasPermissionToChannel(c.Session, c.Params.ChannelId, model.PERMISSION_READ_CHANNEL) {
+		c.SetPermissionError(model.PERMISSION_READ_CHANNEL)
+		return
+	} else {
+		allowed = true
+	}
+
+	if !allowed && !app.SessionHasPermissionToChannel(c.Session, c.Params.ChannelId, model.PERMISSION_MANAGE_SYSTEM) {
+		c.SetPermissionError(model.PERMISSION_MANAGE_SYSTEM)
+		return
+	}
```

Comment:
I think we can remove the `allowed` flag in this check and previous one because I think we can get rid of the PERMISSION_MANAGE_SYSTEM check like I mentioned in the API reference. My fault for including it the spreadsheet